activerecord/CHANGELOG.md

```diff
@@ -1,3 +1,25 @@
+*   Add option to merge association scopes to subqueries.
+
+    It removes a breaking change. Conditions that were not applied in
+    previous versions would be applied now giving people different results.
+
+    Set the option on an association:
+
+    ```ruby
+    class Author < ApplicationRecord
+      has_many :welcome_posts, -> { where(title: "welcome") } , class_name: "Post", apply_on_subqueries: true
```

Comment:
This is the name @rafaelfranca suggested, so let's keep it for now, but I suspect `scope_subqueries`, `apply_scope_on_subqueries`, or similar might be more specific about _what_ it's applying.

I also think the changelog and docs (and original PR description, if I'm being honest) could do with a bit more of a pitch of why you would [or would not] want this to happen. We're currently showing the difference in the generated query, but don't have a prose explanation of the different semantics.